third_party/move/move-compiler-v2/tests/checking/typing/break_any_type.move

```diff
@@ -0,0 +1,20 @@
+module 0x8675309::M {
+    struct Coin {}
+
+    fun t0() {
+        while (true) {
+            0 + break;
```

Comment:
What happens with

const C = break;

?

Reply:
I guess it lets you write 

let x = if (cond) { 0 } else { break; };

okay it makes sense

Reply:
Yes, the above is a good example.

Frankly, I believe all those language design details are just taken over from Rust. I also weren't aware of those and had to consult the Move book for it.

Reply:
PS: however it must be `break` not `break;`